qcodes/tests/dataset/test_subscribing.py

```diff
@@ -50,3 +83,95 @@ def test_basic_subscription(dataset, basic_subscriber):
+                    "factory": "qcodes.tests.dataset.test_subscribing.MockSubscriber",
+                    "factory_kwargs":{
+                        "lg": false
+                    },
+                    "subscription_kwargs":{
+                        "min_wait": 0,
+                        "min_count": 1,
+                        "callback_kwargs": {}
+                    }
+                }
+            }
+        }
+    }
+    """
+    # This little dance around the db_location is due to the fact that the
+    # dataset fixture creates a dataset in a db in a temporary directory.
+    # Therefore we need to 'backup' the path to the db when using the
+    # default configuration.
+    db_location = qcodes.config.core.db_location
+    with default_config(user_config=config):
+        qcodes.config.core.db_location = db_location
+
+        assert 'test_subscriber' in qcodes.config.subscription.subscribers
+
+        xparam = ParamSpec(name='x', paramtype='numeric', label='x parameter',
+                           unit='V')
+        yparam = ParamSpec(name='y', paramtype='numeric', label='y parameter',
+                           unit='Hz', depends_on=[xparam])
+        dataset.add_parameter(xparam)
+        dataset.add_parameter(yparam)
+
+        sub_id = dataset.subscribe(basic_subscriber, min_wait=0, min_count=1,
+                                   state={})
+        sub_id_c = dataset.subscribe_from_config('test_subscriber')
+        assert len(dataset.subscribers) == 2
+        assert list(dataset.subscribers.keys()) == [sub_id, sub_id_c]
+
+        expected_state = {}
+
+        for x in range(10):
+            y = -x**2
+            dataset.add_result({'x': x, 'y': y})
+            expected_state[x+1] = [(x, y)]
+            assert dataset.subscribers[sub_id].state == expected_state
+            assert dataset.subscribers[sub_id_c].state == expected_state
+
+def test_subscription_from_config_wrong_name(dataset):
+    """
+    This test checks that an exception is thrown if a wrong name for a
+    subscriber is passed
+    """
+    # This string represents the config file in the home directory:
+    config = """
+    {
+        "subscription":{
+            "subscribers":{
+                "test_subscriber_wrong":{
+                    "factory": "qcodes.tests.dataset.test_subscribing.MockSubscriber",
+                    "factory_kwargs":{
+                        "lg": false
+                    },
+                    "subscription_kwargs":{
+                        "min_wait": 0,
+                        "min_count": 1,
+                        "callback_kwargs": {}
+                    }
+                }
+            }
+        }
+    }
+    """
+    db_location = qcodes.config.core.db_location
+    with default_config(user_config=config):
+        qcodes.config.core.db_location = db_location
+
+        assert 'test_subscriber' not in qcodes.config.subscription.subscribers
+        with pytest.raises(RuntimeError):
```

Comment:
hope that it's fine now, but I recommend next time to also assert the exception message because there are many RuntimeErrors while we are looking for a very specific one.

Reply:
I agree. I was running out of time and the test should go in before a certain internal release yesterday.